example/convolution2D/src/convolution2D.cpp

```diff
@@ -0,0 +1,381 @@
+/* Copyright 2023 Mehmet Yusufoglu, Bernhard Manfred Gruber, René Widera
+ * SPDX-License-Identifier: ISC
+ */
+
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+#include <iomanip>
+#include <iostream>
+#include <vector>
+
+//! Convolution Example
+//!
+//! A 2D Convolutional filter applied to a matrix. The first kernel, ConvolutionKernel2DGlobalMemory, uses only global
+//! memory. The second kernel ConvolutionKernel2DSharedMemory uses tiling and shared memory. Block size is assumed to
+//! be equal to the tile size. First, the tile is copied to shared memory, since an element in a tile is accessed many
+//! times; using the shared memory for the main matrix data increases performance. Each block works on the domain of
+//! one tile. But at the border of the tile, some external matrix values are needed (at the border with another tile)
+//! then those matrix values are taken from the global memory.
+//! Results can be tested by comparing with the results of the Matlab call: Y =
+//! filter2(FilterMatrix,InputMatrix,'same');
+
+/**
+ * @brief 2D Convolutional Filter using only global memory for the input-matrix and the filter-matrix
+ */
+struct ConvolutionKernel2DGlobalMemory
+{
+    /**
+    @tparam TAcc Accelerator type
```

Comment:
please use all doxygen command starting with `\` in alpaak.

Reply:
done, thanks.